<?xml version="1.0"?>
<launch>
  <node name="extended_toe_finding_and_yolo" pkg="intersection_recognition" type="extended_toe_finding_and_yolo" output="screen" >
    <remap from="scan" to="merged_scan" />
    <remap from="image_raw" to="/image/mercator" />
    <param name="SCAN_HZ" value="10"/>
    <param name="door_size_thresh" value="0.5"/>
    <param name="robot_frame" value="base_link"/>
  </node>

  <node name="scenario_executor" pkg="intersection_recognition" type="scenario_executor" output="screen" />

  <node name="cmd_vel_controller" pkg="intersection_recognition" type="cmd_vel_controller" output="screen" >
    <remap from="cmd_vel" to="icart_mini/cmd_vel" />
    <param name="IMU_HZ" value="100.0"/>
    <param name="reverse_turn" value="1.0"/>
  </node>

<!--
  <node name="emergency_stop" pkg="intersection_recognition" type="simple_emergency_stop" output="screen" >
    <param name="stop_distance_thresh" value="0.2"/>
  </node>
-->

  <node name="scenario_parser" pkg="intersection_recognition" type="scenario_parser.py" output="screen" >
    <param name="scenario_path" value="$(find intersection_recognition)/config/Scenarios/scenario01.txt" />
  </node>

  <node pkg="theta_simple_stitching" type="simple_stitching.py" name="theta_simple_stitching" output="screen">
    <param name="sub_image_topic_name" value="/camera360/image_raw"/>
  </node>


  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find intersection_recognition)/config/intersection_recognition.rviz"/>

</launch>
